vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,539 @@
+In this vignette, we will demonstrate how to use syndromic surveillance system data, particularly the timestamped data captured in each NSSP ESSENCE visit record, to produce nowcasts of cases of a particular diagnosis of interest.
+Note that all visits that originate in the emergency department are used for this analysis, regardless of eventual inpatient admission.
+
+
+## Load packages
+
+We use the `baselinenowcast` package, `dplyr`, and `tidyr` for data manipulation, `readr` for loading in data, `stringr` for parsing text data, `lubridate` for formatting dates, `ggplot2` for plotting, and `purrr` for mapping diagnoses codes to text fields in the data.
+```{r setup, message = FALSE, warning = FALSE, results = 'hide'}
+# Load packages
+library(baselinenowcast)
+library(dplyr)
+library(tidyr)
+library(readr)
+library(stringr)
+library(lubridate)
+library(ggplot2)
+library(purrr)
+# Set seed for reproducibility
+set.seed(123)
+```
+
+
+
+# NSSP data pre-processing
+
+We will walk through how to preprocess the line list NSSP data details dataset in order to obtain a [long tidy dataframe](https://r4ds.had.co.nz/tidy-data.html) containing the incident counts of "cases" of a particular syndrome by reference date (in this instance the date the visit started) and report date (the date the patient's record was updated with the corresponding diagnosis of interest).
+
+## Load in the line list data
+
+This typically will be pulled using an API, but here we provide the `syn_nssp_line_list` dataset as package data.
+```{r}
+syn_nssp_line_list
+```
+<div class="alert alert-warning" role="alert">
+<strong>Note:</strong> This dataset does not represent data from real patients, it is entirely synthetic and designed to mirror the NSSP update fields, which records the timing of a clinical encounter and the corresponding diagnoses code, if any.
+</div>
+
+
+## Define the "syndrome" definition
+
+In this near-real-time emergency department dataset, public health surveillance, or "syndrome" definitions have been developed for a range of public health concerns. These definitions typically rely on the presence of diagnosis code(s), specific free text captured in clinical notes, or a combination of these.
+In some instances, exclusion criteria are incorporated into these definitions to improve their specificity.
+
+In this example, we will focus on syndromic surveillance definitions that rely solely on the presence of one or more diagnoses codes, as this means that patients tend to transition from a non-case to case only once (rather than reverting back to a non-case during the course of their clinical encounter).
+Our nowcasting methods primarily focus on the former use-case, though extensions to be able to account for downward revisions as cases transition from case to non-case, are planned for future work.
```

Comment:
I think you should move this to later when we actually nowcast. It also is more nuanced than this as it can already be done without an extension. I would say when the point in the analysis happens that makes this assumption